### ISSUE TITLE
Improve handling of locales and translations

### ISSUE THREAD

**Issue by mgruner** [zammad/zammad#5184]
### Current State

- A number of locales (= languages) is defined in `locales.yml` and synchronized to the database.
- Some of these have a very low translation completeness (less than 10%), many have a low completeness (< 50%)
- All are shown in the user profile as selectable languages, and in the Knowledge Base as well.
- Admins get a hint if they use a locale with less than 90% completeness that points them to contributing via weblate.

### Use Cases

- As a user, I don't want to be able to select very incomplete translations in my UI as a target language, at least not without any kind of notice about the situation (not only for admins).
- As translation maintainer (@mgruner), I want to remove unmaintained translations from Weblate (after a warning period). The locales can stay in Zammad if they are needed for the KB and not shown in the user profile.
- As knowledge base maintainer, I want to be able to add languages to my system via UI to be able to provide KBs for them, even though they don't exist for the Zammad UI.

**Comment by mantas** [comment]
I'd like to add two points.

Currently we have one option for everything locale-related. UI translation, date & time localization, notifications via channels... It may be nice to bring more granularity in this department. Date & time localization is huge, yet 0.003% of the whole translation :) If we decide to drop shallow translations, we should manage date & time localization in a different way. Something similar to e.g. Mac OS handles translations. UI is not available in smaller languages, yet even small languages have local date & time formats and some small bits like number formatting.

Another issue is that all our translations are in the same bucket. Chat, web form, email/slack/etc notifications, Customer UI, Agent UI, Admin UI... Yet the importance of them are very different. I'd say chat, web and notification templates are top-priority since that's what is visible to the outside world. Customer UI is sort of similar, but I guess vast majority of outside world does not interact with it. Personally I rarely have to log in into some ticket system as a customer. Having those translations alone would be minimum viable translation for many systems with relatively small percentage of the translation. Agent UI is very nice to have with a wee more work. Meanwhile Admin UI is very text-heavy. Yet it's not that important for vast majority of users and use cases. On top of that, in my experience many admins prefer english UI since it makes it easier to follow documentation, tutorials, howtos and so on.

**Comment by downey-lv** [comment]
I agree with the non Admin UI importance, just launched my first Zammad instance and needed a Latvian translation for all parts facing customers and agents.  Latvian sits at an _impressive_ 5% on Weblate (and some seem to be bad machine translations), so had to open two Zammad screens and just go through all of the strings that would be visible. 

Spliiting up the strings could also probably help with getting the Weblate percentages up for the important parts, and also see which languages warrant disabling. I'd love to contribute there, but I'm not sifting through 3k+ unfinished translations and translating everything to Latvian, when I think most (definitely me) would prefer keeping their Admin UI in English.

Also maybe adding Weblate file export from Zammad could help, to share all of the work that has already been done on local translations,

**Comment by MrGeneration** [comment]
> Also maybe adding Weblate file export from Zammad could help, to share all of the work that has already been done on local translations,

I believe that this is already possible. Or I do not understand your point.

![image](https://github.com/zammad/zammad/assets/6549061/42191b26-e646-4992-bd54-0f1738dce383)

**Comment by mgruner** [comment]
Feature wishes from another user: 

> - Separate translations by modules, like: customer portal, admin, system and etc. As our example show, we could have system just in english only, but customer portal we would like to have in 16 languages.
> - Knowledgebase unlimited language selection.

**Comment by downey-lv** [comment]
> > Also maybe adding Weblate file export from Zammad could help, to share all of the work that has already been done on local translations,
> 
> I believe that this is already possible. Or I do not understand your point.

Yes, in Weblate you can export, update and then import the file. I was wondering maybe export from Zammad for import to Weblate would be possible, so you wouldn't have to do the translations twice.

**Comment by AnaelMobilia** [comment]
Hello,

As asked by @mgruner at weblate announce, please found below some feedback based on my experience to translate Zammad:
- "Validation of translation": I translate strings on the development branch of weblate and validate translations. When the translation is backported on the stable weblate branch, the validation status is loss, making it more difficult to identify translations that need to be validated on the stable branch. Suggestion: backport validation status when updating the stable branch.
- "Suggestion deletion": some people provide suggestion on translation. Some of them could be no longer adequate, but as a translater in charge of a language, I can't delete these translation, only accept them. This make more difficult to identify new suggestion.
- "Glossary": some more strings could be added in order to ensure translation quality over time and translater (eg: token, payload, some status, ...)
- False warning about translation quality: Weblate provide false positive on space missing on some case on ruby code on email template. Spaces are needed in French before "?" and ":" keyword, but when in ruby, it's not a string word, but a ruby keyword. ([example](https://translations.zammad.org/translate/zammad/zammad-development/fr/?q=%3C%25+if+%40objects%5B%3Achanges%5D.present%3F+%25%3E&sort_by=-priority%2Cposition&checksum=#nearby))
- "Source string location": it always refer to the line 0 of the file, which implies to search the string again in the file. Is it possible to have a direct link to the line?
- Technical strings: some strings are really hard to translate ([example](https://translations.zammad.org/translate/zammad/zammad-development/fr/?checksum=692b8441b857a6f9)). It's usually keywords list (separated by commas). Is it needed to be translated or not ?

Bests regards,
Anael

**Comment by mgruner** [comment]
Dear @AnaelMobilia,

Thank you so much for your valuable feedback. 

> "Validation of translation": I translate strings on the development branch of weblate and validate translations. When the translation is backported on the stable weblate branch, the validation status is loss, making it more difficult to identify translations that need to be validated on the stable branch. Suggestion: backport validation status when updating the stable branch.

I don’t have a solution for this one. My suggestion would be to only review translations on develop, as it is the future stable version.

> "Suggestion deletion": some people provide suggestion on translation. Some of them could be no longer adequate, but as a translater in charge of a language, I can't delete these translation, only accept them. This make more difficult to identify new suggestion.

I just added the permissions to manage suggestions to the Reviewers group, so it should not work for you, can you try please?

> "Glossary": some more strings could be added in order to ensure translation quality over time and translater (eg: token, payload, some status, ...)

You should be able to add strings to the French glossary with the Reviewers permissions that you have. Does it not work?

> False warning about translation quality: Weblate provide false positive on space missing on some case on ruby code on email template. Spaces are needed in French before "?" and ":" keyword, but when in ruby, it's not a string word, but a ruby keyword. ([example](https://translations.zammad.org/translate/zammad/zammad-development/fr/?q=%3C%25+if+%40objects%5B%3Achanges%5D.present%3F+%25%3E&sort_by=-priority%2Cposition&checksum=#nearby))

Right. I disabled this check, which conflicts with our ERB templates.

> "Source string location": it always refer to the line 0 of the file, which implies to search the string again in the file. Is it possible to have a direct link to the line?

Added, hope it is useful!

> Technical strings: some strings are really hard to translate ([example](https://translations.zammad.org/translate/zammad/zammad-development/fr/?checksum=692b8441b857a6f9)). It's usually keywords list (separated by commas). Is it needed to be translated or not ?

Yes please. They are used to filter/find profile settings in the new desktop view.

**Comment by AnaelMobilia** [comment]
Hi @mgruner 

Thanks for the update and actions!

> > "Suggestion deletion": some people provide suggestion on translation. Some of them could be no longer adequate, but as a translater in charge of a language, I can't delete these translation, only accept them. This make more difficult to identify new suggestion.
> 
> I just added the permissions to manage suggestions to the Reviewers group, so it should not work for you, can you try please?

Perfect!
 
> > "Glossary": some more strings could be added in order to ensure translation quality over time and translater (eg: token, payload, some status, ...)
> 
> You should be able to add strings to the French glossary with the Reviewers permissions that you have. Does it not work?

Thanks!
Additional question: is it possible to have an alert for translation which didn't respect the glossary?
My idea is to use it as a guideline and found translations which didn't "meet" the glossary (with the possibility to ignore the warning)

Bests regards,
Anael

**Comment by mgruner** [comment]
> Thanks! Additional question: is it possible to have an alert for translation which didn't respect the glossary? My idea is to use it as a guideline and found translations which didn't "meet" the glossary (with the possibility to ignore the warning)

I am not aware of such a feature in Weblate, sorry.

**Comment by fastone2030** [comment]
Hi aneal,

Thank for your mail please contact me our my number 0966541732425

(Issue #5184)

Regards,


***@***.***?anonymous&ep=bwmEmailSignature>
Book time to meet with ***@***.***?anonymous&ep=bwmEmailSignature>

**Comment by JPBRM** [comment]
Feedback based on the question asked translators on 5-6: sometimes the context of a translation is unclear. Even after looking a the source code it is hard to figure out where the string will be shown within zammad. Recently I also started translating immich, and the developers of immich help translators by adding screenshots of the original language within the system. An example can be found here: https://hosted.weblate.org/translate/immich/immich/nl/?&offset=4 and here https://hosted.weblate.org/translate/immich/immich/nl/?&offset=10

This really helps when translating new functionality not available in the current release and how the text wil be shown in the system.

**Comment by gwingend** [comment]
Guten Tag,

vielen Dank für Ihre E-Mail.
Ich bin nicht im Büro.
Ihre Mail wird während meiner Abwesenheit weder gelesen noch weitergeleitet.

Bitte wenden Sie sich bei technischen Problemen an den EDV-Helpdesk ***@***.***).



Mit freundlichen Grüßen
Gerhard Wingender
Amt für IT

>>> JPBRM ***@***.***> 31.7.24 17:12 >>>

Feedback based on the question asked translators on 5-6: sometimes the context of a translation is unclear. Even after looking a the source code it is hard to figure out where the string will be shown within zammad. Recently I also started translating immich, and the developers of immich help translators by adding screenshots of the original language within the system. An example can be found here: https://hosted.weblate.org/translate/immich/immich/nl/?&offset=4 and here https://hosted.weblate.org/translate/immich/immich/nl/?&offset=10

This really helps when translating new functionality not available in the current release and how the text wil be shown in the system.



-- 
Reply to this email directly or view it on GitHub:
https://github.com/zammad/zammad/issues/5184#issuecomment-2260758646
You are receiving this because you are subscribed to this thread.

Message ID: ***@***.***>
-- 
Diese E-Mail wurde von Sophos UTM auf Viren überprüft.

Besuchen Sie uns im Internet unter: www.neuwied.de
oder bei Facebook @neuwied.de
und Instagram @deichstadt_neuwied

Eine E-Mail unterliegt dem Kommunikationsgeheimnis und ist nur für den Empfänger bestimmt. Der Inhalt ist grundsätzlich als vertraulich zu betrachten und nicht für Dritte zugänglich. Eine Weiterleitung bedarf daher der Zustimmung des Verfassers. Diese E-Mail darf zu dienstlichen Zwecken innerhalb der Stadtverwaltung weitergeleitet werden.

Sparen Sie pro Seite ca. 200 ml Wasser, 2 g CO2 und 2 g Holz: Drucken Sie daher bitte nur, wenn es wirklich notwendig ist!

**Comment by Mant1kor** [comment]
> * As a user, I don't want to be able to select very incomplete translations in my UI as a target language, at least not without any kind of notice about the situation (not only for admins).

In other projects, I've seen they just add percentage of completeness to the language selector. Weblate has this, if you could use it — it would be good enough hint for users and admins. 
And I think showing some languages in native form and some in form "english name(native name)" is useless and inconsistent. Only native names are better. As a user, I know all languages native names I'm interested in.
P.S. Have you considered switching to the Crowdin translation platform? As a translator, I think it's way more comfortable to work with.

![Image](https://github.com/user-attachments/assets/6f2ab822-3c1b-441a-8dcc-8b9390eb67bf)
